content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -668,27 +668,29 @@ consume.
 
 ## Job patterns
 
-The Job object can be used to support reliable parallel execution of Pods. The Job object is not
-designed to support closely-communicating parallel processes, as commonly found in scientific
-computing. It does support parallel processing of a set of independent but related *work items*.
-These might be emails to be sent, frames to be rendered, files to be transcoded, ranges of keys in a
-NoSQL database to scan, and so on.
+The Job object can be used to process a set of independent but related *work items*.
+These might be emails to be sent, frames to be rendered, files to be transcoded,
+ranges of keys in a NoSQL database to scan, and so on.
 
 In a complex system, there may be multiple different sets of work items. Here we are just
 considering one set of work items that the user wants to manage together &mdash; a *batch job*.
 
 There are several different patterns for parallel computation, each with strengths and weaknesses.
 The tradeoffs are:
 
-- One Job object for each work item, vs. a single Job object for all work items. The latter is
-  better for large numbers of work items. The former creates some overhead for the user and for the
-  system to manage large numbers of Job objects.
+- One Job object for each work item, vs. a single Job object for all work items. The former creates
+  some overhead for the user and for the system to manage large numbers of Job objects. The latter
+  is better for large numbers of work items. 
 - Number of pods created equals number of work items, vs. each Pod can process multiple work items.
   The former typically requires less modification to existing code and containers. The latter
   is better for large numbers of work items, for similar reasons to the previous bullet.
 - Several approaches use a work queue. This requires running a queue service,
   and modifications to the existing program or container to make it use the work queue.
   Other approaches are easier to adapt to an existing containerised application.
+- When the Job is accompanied with a
+  [headless Service](/docs/concepts/services-networking/service/#headless-services)
```

Comment:
This is necessary but not necessarily sufficient I think.

Reply:
anything in particular?
The details are in Job with Pod-to-Pod communication, linked in the table below

Reply:
It might be out of scope of talking about solely the Job structure - that in and of itself providing the hostnames is the start for what you need for a scientific computing workflow. But if you mean HPC when you say scientific computing, the networking fabric backing the connectivity between pods is everything. I think (for this document) I would add a brief note about that, something like:

> you can enable the processes within a Job to communicate, as commonly necessary in scientific computing. The latency and bandwidth for this communication that could determine the success of the workflow, while essential, are out of scope of the design of the Job itself.

Reply:
Indeed, this document is about the job/pod structure.
In the past, the hostnames were not predictable, so HPC was close to impossible.

I think notes about latency are beyond the scope. You can imagine that you might have similar limitations writing certain kinds of tradicional non-HPC services, depending on your domain.

So I'm thinking of leaving this as-is. Unless sig-docs has another suggestion?

Reply:
Understood! I thought I'd mention it because the title explicitly mentions scientific computing. No worries if it's out of scope.

Reply:
Maybe this would be worth a post on its own @vsoch?  I know you have gone through setting up HPC-Like environments on some of the cloud vendors.  I'd think that would make an interesting post on the k8 blog but I think that would require sig-docs thoughts.

Reply:
As long as it's vendor neutral, it would be a great fit for a blog post. Although I think it might be hard to do so.

Reply:
By vendor neutral I mean that the samples and steps that you present should probably be able to run anywhere.

Reply:
The approaches for low latency on other clouds have (so far) required specialized hardware, so the difference alone in hardware options might make that impossible.

I did see the new (CDI?) I think it was called in the new Kubernetes release - I wonder if that could be a direction?

Reply:
@vsoch if you put a nice warning, that the exercises mentioned in the post were executed on a self-hosted infra, due to hardware requirement, but describing all the details would be very beneficial for the broader community running into similar problems like you have. That kind of approach would also mean it's vendor natural, b/c I assume based on what you wrote that you basically are running DIY k8s, right?